### PR TITLE
Chrisk bugfixes

### PR DESCRIPTION
**Pull request by chriskr**

### CODE REVIEW

File: src/lib/storageprototype.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  var raw_value = this.getItem(key);
+  var value = null;
+  try
+  {
+    value = JSON.parse(raw_value);
+  }
+  catch(e)
+  {
+    value = default_value;
+  };
```
Comment (hzr) [inline src/lib/storageprototype.js]: Extra semicolon.

File: src/lib/storageprototype.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return value === null || value === undefined ? default_value : value;
```
Comment (hzr) [inline src/lib/storageprototype.js]: `return value == null ? default_value : value;` would suffice, but this is fine.

File: src/ui-scripts/settings.js [hunk 3]
```diff
@@ -32,12 +31,10 @@ var SettingsBase = function()
    */
   this.get = function(key)
   {
-    var val = "";
-    return (
-    typeof this.map[key] !== 'undefined' ?
-    this.map[key] :
-    (this.map[key] = ((val = window.localStorage.getItem(key)) ? JSON.parse(val) : null)));
-  }
+    if (typeof this.map[key] == "undefined")
```
Comment (hzr) [inline src/ui-scripts/settings.js]: No need for `typeof` here. And if that disappears, maybe we can check against `null` (to catch that too if needed):

``` javascript
if (this.map[key] == null)
```

**Comment by hzr** [comment]
Looks fine otherwise.

**Comment by danfooo** [comment]
Looks fine, I didn't look deep enough to know what checks are really needed where, but there are comments on that already.

File: src/lib/storageprototype.js (hunk 2 above)
Comment (p01) [inline src/lib/storageprototype.js]: Agreed.

**Comment by p01** [comment]
All good beside the null/undefined checks that can be combined.

**Comment by chriskr** [comment]
Simplified boolean check.

**Comment by hzr** [comment]
Looks good!

**Pull request merged**